frontend/src/survey/Survey.jsx

```diff
@@ -0,0 +1,180 @@
+import {useContext, useEffect, useRef, useState} from "react";
+import { SurveyContext } from "../context/SurveyContext.jsx";
+import { UserContext } from "../context/UserContext.jsx";
+import { InputTextarea } from 'primereact/inputtextarea';
+import {RadioButton} from "primereact/radiobutton";
+import {Card} from "primereact/card";
+import {InputNumber} from "primereact/inputnumber";
+import {Toast} from "primereact/toast";
+import {Button} from "primereact/button";
+import {submitSurvey} from "../api/surveyApi.js";
+import {InputText} from "primereact/inputtext";
+import {FloatLabel} from "primereact/floatlabel";
+import {Message} from "primereact/message";
+
+
+/*
+The Survey component models a Survey that the user can fill
+out and submit.
+ */
+const Survey = () => {
+    // Get reference to survey and user context
+    const [survey] = useContext(SurveyContext);
+    const [user] = useContext(UserContext);
+
+    // useState hook for setting/getting user's email address (if not logged in)
+    const [email, setEmail] = useState('');
+
+    // Sort the array of questions based on questionId
+    const surveyQuestions = survey.questions;
+    surveyQuestions.sort((a, b) => a.questionId - b.questionId);
+
+    const toast = useRef(null);
+
+    // Create useState hook to keep track of the user's response for each question
+    // for each question using a Map -> [questionId, response] is the key-value pair
+    const [responses, setResponses] = useState(new Map(surveyQuestions.map((question) => [question.questionId, undefined])));
+
+    // Handle response change by updating the Map with the new value for the key
+    const handleResponseChange = (e, questionId) => {
+        const updatedResponse = e.target.value;
+        // Update the response value for the question and create new Map to force re-render
+        setResponses(new Map(responses.set(questionId, updatedResponse)));
+    }
+
+    // Hook to update the email with the logged-in user's email
+    useEffect(() => {
+        if (user) {
+            setEmail(user.email);
+        }
+    }, [user]);
+
+    // Handling clicking of submit button
+    const surveySubmit = async () => {
+        // Set the user's email if they are logged in
+        if (user) {
+            setEmail(user.email);
+        }
+        // Check email field
```

Comment:
I've added a function in src/modules/validation.js that verifies the format of an email through a regular expression. Can also run the test here and notify if the email is invalid and prevent the survey from being sent.

Reply:
Yeah good catch. The call to validateEmail() has now been added.